#!/usr/bin/env python
PACKAGE = "iiwa_toolkit"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# DS Gains
gen.add("Position_DSgain",   double_t,    0, "scalling the preset ds gain from yaml", 1,  0.5, 2)
gen.add("Orientation_DSgain",   double_t,    0, "scalling the preset ds gain from yaml", 1,  0.5, 2)

# Passive Gains
gen.add("Position_lambda",   double_t,    0, "scalling the preset lambda0 and lambda1 from yaml", 1,  0.5, 2)
gen.add("Orientation_lambda",   double_t,    0, "scalling the preset lambda0 and lambda1 from yaml", 1,  0.5, 2)

# Desired Positon
gen.add("dX_des",   double_t,    0, "displacing the preset des_position from yaml", 0,  -0.25, 0.25)
gen.add("dY_des",   double_t,    0, "displacing the preset des_position from yaml", 0,  -0.5, 0.5)
gen.add("dZ_des",   double_t,    0, "displacing the preset des_position from yaml", 0,  -0.5, 0.5)

# Desired Orientation
gen.add("dX_des_angle",   double_t,    0, "displacing the preset des_orientation from yaml", 0,  -1.5, 1.5)
gen.add("dY_des_angle",   double_t,    0, "displacing the preset des_orientation from yaml", 0,  -1.5, 1.5)
gen.add("dZ_des_angle",   double_t,    0, "displacing the preset des_orientation from yaml", 0,  -1.5, 1.5)

# Inertia parameters
gen.add("Inertia_gain",   double_t,    0, "change the inertia control multiplier", 0.001,  0, 1.0)
gen.add("Inertia_desired",   double_t,    0, "change the desired inertia", 5.0,  0.0, 10.)


exit(gen.generate(PACKAGE, "iiwa_toolkit", "passive_cfg_params"))